rpcs3/Emu/RSX/Common/ProgramStateCache.cpp

```diff
@@ -205,34 +205,34 @@ vertex_program_utils::vertex_program_metadata vertex_program_utils::analyse_vert
 	{
 		for (u32 i = instruction_range.first, count = 0; i <= instruction_range.second; ++i, ++count)
 		{
-			const qword* instruction = reinterpret_cast<const qword*>(&data[i * 4]);
-			qword* dst = reinterpret_cast<qword*>(&dst_prog.data[count * 4]);
+			const u32* instruction = &data[i * 4];
+			u32* dst = &dst_prog.data[count * 4];
 
 			if (result.instruction_mask[i])
 			{
-				dst->dword[0] = instruction->dword[0];
-				dst->dword[1] = instruction->dword[1];
+				auto inst = v128::loadu(instruction);
```

Comment:
Did you confirm that this loadu/storeu is a no-op? In most cases where it is used in this file it can be easily replaced with direct access to instruction[x] if not, and it would still be easily readable. e.g instruction is already u32* so `d2.HEX = instruction[2]` is still just as readable.

Reply:
IT's not no-op, the store was moved to be after the if statement so it won't need to touch dst specfically later.

Reply:
Then how is it better than just declaring instruction as u64* and indexing indo it?
Same goes for the other parts of the code. If it is worse then we shouldn't use it.